app/models/message.rb

```diff
@@ -0,0 +1,102 @@
+require 'aws-sdk-sqs'
+
+class Message
+  extend ActiveModel::Naming
+  include ActiveModel::Validations
+  include ActiveModel::Model
+  attr_accessor :barcodes, :user_email, :protect_cgd, :action, :bib_record_number
+  
+  validate :barcode_format, :bib_record_number_format
+
+  def valid_barcodes
+    if barcodes && barcodes.is_a?(Array)
+      barcodes.map!(&:to_s) 
+      barcodes.select{ |b| b[/^\d{14}$/] }
+    else
+      []
+    end
+  end
+  
+  def invalid_barcodes
+    invalid = []
+    if barcodes && barcodes.is_a?(Array)
+      invalid = barcodes 
+      invalid = invalid - valid_barcodes
+    end
+    return invalid
+  end
+  
+  def send_update_message_to_sqs
+    sqs = Aws::SQS::Client.new(
+      region: 'us-east-1',
+      access_key_id: AWS_KEY_ID,
+      secret_access_key: AWS_SECRET
+    )
+    queue_url = SQS_QUEUE_URL
+    entries = [
+        {
+          id: self.barcodes.first,
+          message_body: "{
```

Comment:
Two things:

1.  If we are already storing stuff from the OAuth response in the session, you can probably get the current user's email for this.

2.  If you want to get out of hand-rolling JSON....MAYYYBEEEE this will take a hash and JSONify it OR.....
call JSON.generate({action: self.action, barcodes: self.barcodes})